backend/src/main/java/de/bund/digitalservice/ris/caselaw/domain/HandoverService.java

```diff
@@ -85,19 +89,37 @@ public XmlTransformationResult createPreviewXml(UUID documentUuid)
     return mailService.getXmlPreview(documentationUnit);
   }
 
+  /**
+   * Create a preview juris xml for a edition
+   *
+   * @param editionId the id of the edition
+   * @return the export result, containing the juris xml and export metadata
+   */
+  public List<XmlTransformationResult> createEditionPreviewXml(UUID editionId) throws IOException {
+    LegalPeriodicalEdition edition =
+        editionRepository
+            .findById(editionId)
+            .orElseThrow(() -> new IOException("Edition not found: " + editionId));
+    return mailService.getXmlPreview(edition);
+  }
+
   /**
    * Get the event log for a documentation unit, containing jDV email handover operations, handover
    * reports (response emails from the jDV) and migrations/import events.
    *
-   * @param documentUuid the UUID of the documentation unit
+   * @param entityId the UUID of the documentation unit
    * @return the event log
    */
-  public List<EventRecord> getEventLog(UUID documentUuid) {
+  public List<EventRecord> getEventLog(UUID entityId, HandoverEntityType entityType) {
+    if (entityType == HandoverEntityType.EDITION) {
```

Comment:
For clarity, it seems like we can use switch cases:

case (DOCUMENTATION_UNIT)...
case(Edition)....

I usually not a fan of switch for two cases only but comined with enum it makes sense

Reply:
Good idea! I changed it that way!